Jenkinsfile.tap

```diff
@@ -1,63 +1,54 @@
-}
-
 
-ansiColor('xterm') {
-  timestamps {
-    timeout(60) {
-      node('NODE_JS_BUILDER') {
-        try {
-          stage('Install') {
-            withCredentials([
-              string(credentialsId: 'NPM_TOKEN', variable: 'NPM_TOKEN')
-            ]) {
-              sh 'echo \'//registry.npmjs.org/:_authToken=${NPM_TOKEN}\' > $HOME/.npmrc'
-              sh '''#!/bin/bash -ex
-              source ~/.nvm/nvm.sh
-              nvm install 7
-              nvm use v7
-              npm install
-              rm -f $HOME/.npmrc
-              '''
-            }
-          }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    timestamps()
+    ansiColor('xterm')
+  }
 
-          stage('Test') {
-            echo "Starting Tap Tests"
-            try {
-              tapTest()
-            }
-            catch(error) {
-              echo "Test Failed"
-              sleep 30s
-              tapTest()
-            }
-            echo "Finised with Tap Tests"
-          }
+  stages {
+    stage('Install') {
+      steps {
+        withCredentials([
+          string(credentialsId: 'NPM_TOKEN', variable: 'NPM_TOKEN')
+        ]) {
+          sh 'echo \'//registry.npmjs.org/:_authToken=${NPM_TOKEN}\' > $HOME/.npmrc'
+          sh '''#!/bin/bash -ex
+          source ~/.nvm/nvm.sh
+          nvm install 7
+          nvm use v7
+          npm install
+          rm -f $HOME/.npmrc
+          '''
         }
+      }
+    }
 
-
-        catch(error) {
-          if (currentBuild.result != 'UNSTABLE') {
-            currentBuild.result = 'FAILURE'
+    stage('Test') {
+      steps {
+        withCredentials([
+          string(credentialsId: 'ddfd04fb-e00a-4df0-9250-9a7cb37bce0e', variable: 'CISCOSPARK_CLIENT_SECRET'),
+          usernamePassword(credentialsId: 'SAUCE_LABS_VALIDATED_MERGE_CREDENTIALS', passwordVariable: 'SAUCE_ACCESS_KEY', usernameVariable: 'SAUCE_USERNAME'),
+        ]) {
+          retry(2){
+            sh '''#!/bin/bash -ex
+            source ~/.nvm/nvm.sh
+            nvm use v7
+            CISCOSPARK_CLIENT_ID=C873b64d70536ed26df6d5f81e01dafccbd0a0af2e25323f7f69c7fe46a7be340 SAUCE=true npm run test:tap
+            '''
           }
-          cleanup()
-          emailext body: '<p>Spark-js-sdk--react-widget--tap build number $BUILD_NUMBER has failed. For detailed information, please refer to $BUILD_URL</p><pre>${BUILD_LOG_EXCERPT, start="^Starting Tap Tests", end="^Finised with Tap Tests"}</pre>', subject: 'Spark-js-sdk--react-widget--tap Failure', to: 'spark-js-sdk-gating-pipeline-email.94d222a7@ciscospark.pagerduty.com'
-          throw error
         }
       }
     }
   }
+
+  post {
+    failure {
+      echo "build failed"
+      //emailext attachLog: true, body: 'Spark-js-sdk--react-widget--tap build number $BUILD_NUMBER has failed. For detailed information, please refer to $BUILD_URL', subject: 'Spark-js-sdk--react-widget--tap Failure', to: 'spark-js-sdk-gating-pipeline-email.94d222a7@ciscospark.pagerduty.com'
```

Comment:
Looks like pagerduty email is commented out? Also could you test the formatter we added in the previous version? It should at least provide us with text in pager duty that tells us why things failed.

Reply:
Good catch, missed this. Lex probably had it commented out from when he was testing. As far as the formatter goes, I will have to ask about the priority of that as I'm still working with the dev ex team on things they need.

Reply:
Check the previous version of the jenkinsfile. You should be able to copy it over and run it once to make sure there are no issues.

Reply:
Okay, will test it and then get a second CY approval.